QuickStart/FAQBot/FAQBotMessageCell.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  FAQBotMessageCell.swift
+//  Quickstart
+//
+//  Created by Jaesung Lee on 2021/12/19.
+//
+
+import UIKit
+import SendBirdDesk
+
+protocol FAQBotMessageCellDelegate: AnyObject {
+    func didSelectQuestion(_ faqFileID: Int64, question: String)
+}
+
+class FAQBotMessageCell: UITableViewCell {
+    static let identifier = "FAQBotMessageCell"
+    @IBOutlet weak var profileImageView: UIImageView!
+    @IBOutlet weak var faqStackView: UIStackView!
+    @IBOutlet weak var dateLabel: UILabel!
+    
+    
+    var faqData: SBDSKFAQData?
+    weak var delegate: FAQBotMessageCellDelegate?
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        // Initialization code
+    }
+
+    override func setSelected(_ selected: Bool, animated: Bool) {
+        super.setSelected(selected, animated: animated)
+
+        // Configure the view for the selected state
+    }
+    
+    func configure(message: SBDBaseMessage, faqData: SBDSKFAQData) {
+        self.faqData = faqData
+        if let url = URL(string: message.sender?.profileUrl ?? "") {
+            profileImageView.load(url: url)
+        }
+        
+        let timeFormat = "hh:mm"
+        dateLabel.text = Date
+            .from(message.createdAt)
+            .sbu_toString(formatString: timeFormat)
+        
+        // Create FAQ message views and add to the stack view
+        faqStackView.arrangedSubviews.forEach {
+            $0.removeFromSuperview()
+        }
+        faqData.faqResults.forEach { result in
+            let faqMessageView = FAQMessageView(result: result)
+            faqMessageView.frame = CGRect(x: 0, y: 0, width: 244, height: 82)
+            faqMessageView.delegate = self
+            faqStackView.addArrangedSubview(faqMessageView)
+        }
+        faqStackView.updateConstraints()
+    }
+}
+
+extension FAQBotMessageCell: FAQMessageViewDelegate {
+    func didSelect(_ question: String) {
+        guard let faqFileID = faqData?.faqFileId else { return }
+        delegate?.didSelectQuestion(faqFileID, question: question)
+    }
```

Comment:
Could we have just ChatViewController implement this FAQMessagViewDelegate, and only FAQMessageView calls this delegate method? It feels unnecessary to have FAQBotMessageCell implement this delegate and relay it to ChatViewController again as a delegate.

Reply:
The, FAQMessageView must have FAQ data which has all information of FAQ in the message such as a list of FAQ results and FAQ file ID. However, IMO, FAQMessageView is a view for FAQ result only, so it just needs a FAQ result which is included in FAQ data, not all information of FAQ.